File: app/src/main/java/eugeen3/keepinfit/viewmodels/FoodItemViewModel.java
package eugeen3.keepinfit.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import eugeen3.keepinfit.entities.FoodItem;
import eugeen3.keepinfit.repositories.FoodItemRepository;

public class FoodItemViewModel extends AndroidViewModel {
    private FoodItemRepository repository;
    private LiveData<List<FoodItem>> allFoodItems;

    public FoodItemViewModel(@NonNull Application application) {
        super(application);
        repository = new FoodItemRepository(application);
        allFoodItems = repository.getAllFoodItems();
    }

    public void insert(FoodItem foodItem) {
        repository.insert(foodItem);
    }


    public void update(FoodItem foodItem) {
        repository.update(foodItem);
    }


    public void delete(FoodItem foodItem) {
        repository.delete(foodItem);
    }


    public void deleteAllFoodItems() {
        repository.deleteAllFoodItems();
    }

    public LiveData<List<FoodItem>> getAllFoodItems() {
        return allFoodItems;
    }
}

File: app/src/main/java/eugeen3/keepinfit/viewmodels/FoodItemViewModel.kt
package eugeen3.keepinfit.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import eugeen3.keepinfit.repositories.FoodItemRepository
import androidx.lifecycle.LiveData
import eugeen3.keepinfit.entities.FoodItem

class FoodItemViewModel(application: Application) : AndroidViewModel(application) {
    val repository: FoodItemRepository = FoodItemRepository(application)
    val allFoodItems: LiveData<List<FoodItem>>

    fun insert(foodItem: FoodItem?) {
        repository.insert(foodItem)
    }

    fun update(foodItem: FoodItem?) {
        repository.update(foodItem)
    }

    fun delete(foodItem: FoodItem?) {
        repository.delete(foodItem)
    }

    fun deleteAllFoodItems() {
        repository.deleteAllFoodItems()
    }

    init {
        allFoodItems = repository.allFoodItems!!
    }
}